### ISSUE TITLE
As a Concourse operator with many pipelines, I would like to move a pipeline from the "main" team to a new team

### ISSUE THREAD

**Issue by SocalNick** [concourse/concourse#1164]
# Feature Request

## What challenge are you facing?

We have a concourse installation with many pipelines, most of which, our "main" team is not really responsible for. We've given a lot of Github Org/Teams access to our concourse, but we are ready to take advantage of concourse teams. We can create the team, but the existing pipelines would have to be deleted and re-created with the new team, losing all our precious build history.

## A Modest Proposal

Maybe, `fly set-pipeline-team -p some-pipeline -n some-team`

**Comment by chendrix** [comment]
I think I'm going to try expanding the scope of this story a bit to be able for any team to move a pipeline to any other team.

I think the biggest UX implications are confirmation that you really want to do that, and obviously that a new pipeline will just show up in some new team, potentially exposing credentials to a different team.

**Comment by SocalNick** [comment]
Awesome, thanks @chendrix !

**Comment by chendrix** [comment]
After talking with @ebabani and @jtarchie there's a potentially malicious security attack if you can assign a pipeline (e.g. one that's a memory bomb) and it starts automatically on team-specific or tagged workers.

**Comment by SocalNick** [comment]
So limit moving pipelines from main team to other teams?

**Comment by marco-m** [comment]
@chendrix maybe I am missing something, but the root cause of this "potentially malicious attack" is because allowing teamA to move a pipeline to teamB actually gives teamA "write access" to teamB.

Said in another way: although a member of teamA cannot create a pipeline directly in teamB, it can still do it by "moving the pipeline". 

As such, there are only two secure ways to implement this feature:

1. Make it a two-step process: teamA moves a pipeline to teamB and the pipeline goes into a "wait for acceptance" mode. Then a member of teamB has to explicitly "accept" the pipeline. This would be safe but to me looks like complicated.
2. Only somebody with write access to both teams can move the pipeline, or somebody that belongs to the main team, which is already the de-facto admin team.

**Comment by SocalNick** [comment]
The original feature request was moving a pipeline from main to another team. If we consider members of the main team administrators, that seems to alleviate the security concerns.

**Comment by marco-m** [comment]
@SocalNick yes, sorry I should have been more explicit. Option 2 would, in my understanding, cover your initial request with no additional implementation work.

**Comment by willejs** [comment]
I think I managed to do this before by manipulating the postgres database if you cant wait.

**Comment by willejs** [comment]
@SocalNick I just wrote a gist with a few examples on what to do in the db [here](https://gist.github.com/willejs/7a58cd74f1b5da7d1a5bb83f2dbf8d2e)

**Comment by amuessig** [comment]
Any progress on this?

**Comment by professor** [comment]
We just upgraded from `3.x` to `4.2.2` and felt this pain. We do not want to loose history of our pipelines when moving them from `main` to another team.

**Comment by Savemech** [comment]
Any updates on this?

**Comment by vito** [comment]
@Savemech Nope - happy to review a PR for it.

I'm a little iffy on putting it in `set-pipeline` - changing the ownership of a pipeline feels like a discrete separate action similar to `rename-team` or `rename-pipeline`. What about `chown-pipeline`?

I agree with @marco-m that checking that the user has write access to both teams should be safe enough. That would have been difficult before but now that users log in to all their teams at once this should be pretty easy. :slightly_smiling_face: /cc @pivotal-jwinters

**Comment by marknet15** [comment]
I missed that this issue was already open which led to me creating #4246 my main use case was around retaining all build history and resources data etc when the pipeline is moved from one team to another for example `main` to another `new-main`. So this is some functionality that would be great to have

**Comment by SocalNick** [comment]
Just found this lingering at the bottom of my open GH issues. I haven't used Concourse since leaving Pivotal in 2019, closing. Hope everyone is well!

**Comment by RichardBradley** [comment]
@SocalNick -- while you may no longer need this, I think quite a few other people would like to see this feature and are following this ticket (22 thumbs up reacts on the top).

Would you mind reopening this, please?

**Comment by marco-m-pix4d** [comment]
Just re-stumbled upon this because I now need it. Re-opening, as suggested by @RichardBradley.

**Comment by Guyverix** [comment]
I am stuck needing this now as well.

**Comment by taylorsilva** [comment]
Anyone is free to submit a PR adding this feature
